### PR TITLE
Fixed UI Inconsistencies while quoting messages

### LINKED ISSUE #563: [BUG] UI inconsistencies in quoted messages
### Steps to reproduce:

1. Setup Embedded Chat
2. Quote a message and send it, compare it with RC (UI has to be fixed and match with RC)
3. Now Quote a quoted message and do it recursively, the UI in EC will fail to render properly. In RC, 3 recursive messages can be quoted and 2 recent ones will be displayed on screen without fail.
4. Try to quote a image / video attachment, no preview is available

### Expected behavior:

UI consistency while quoting messages
Even in case of recursive quoted messages, the UI should render correctly (Refer to RC)
In case of attachment messages quoted, the preview should display correctly (In video, i have shown w.r.t image, test it with multiple attachments and ensure that everything is rendering properly.
 
### Actual behavior:

UI inconsistencies are present and UI fails to render properly when recursive messages are quoted.

https://github.com/RocketChat/EmbeddedChat/assets/78961432/e837052c-3adb-4c2e-8b34-4091ef840e52

Note: I think the code is written in PinnedAttachment.js for rendering, make sure it must not break or change pinned message notification UI

Edit:  The same has to be handled in QuoteAttachment preview as well.

https://github.com/RocketChat/EmbeddedChat/assets/78961432/e947d566-5a9e-41f3-b123-cd4f519b3390

### PR DESCRIPTION
**Pull request by devanshkansagra**
## Acceptance Criteria fulfillment

- [x] Added support for quoting different types of attachements
- [x] Implemented displaying of attachement while quoting 
- [x] Fix recursive quoting
- [x] Fix design of quoted message in curved variants

Fixes #563 

## Video/Screenshots
[Screencast from 2024-07-14 16-41-33.webm](https://github.com/user-attachments/assets/3dabb53c-69d1-40fe-8690-e7674badd872)

### CODE REVIEW

**Comment by Spiral-Memory** [comment]
Hey @devanshkansagra,

Thanks for the contribution.

While quoting messages, there's a sideline that appears and quotes the message. It seems like you simply removed it, which shouldn't be the case. Please also try it in Storybook so that we can see the changes across all design variants. It must work properly in all variants. Additionally, I don't think it's a good idea to show thumbnails in "QuoteBox." Images look fine, but video and audio appear way too big.

**Comment by devanshkansagra** [comment]
Okay will check that

**Comment by devanshkansagra** [comment]
image

**Comment by devanshkansagra** [comment]
Hey maintainers, I have fixed the issue of recursive quoting, but I stuck at styling of recursive quoting of messages in curved variants, I have explored a lot, but I didn't get any clue. So anyone could please suggest me, what kind of design improvements I need to make to fix this

This is the design issue in curved variant

image

It looks proper in other variants (Ex StormySeas)

image

**Comment by Spiral-Memory** [comment]
Hey @devanshkansagra  It looks good !

Let me know the exact problem you are facing in fixing the design in curved variant as well

**Comment by Spiral-Memory** [comment]
Also please update the video with your latest changes

**Comment by devanshkansagra** [comment]
> Hey @devanshkansagra It looks good !
> 
> Let me know the exact problem you are facing in fixing the design in curved variant as well

Most basically the improper alignment

**Comment by devanshkansagra** [comment]
> Also please update the video with your latest changes

Updated the video

**Comment by Spiral-Memory** [comment]
> > Also please update the video with your latest changes
> 
> Updated the video

Checked your video, Great work! 

Can we fix this as well before fixing for the curved variant, that it should display the actual message here instead of msg link:

image

Also, in recursive quoting, the video / image is not visible, I know you have let me know about it that even with main RC it doesn't work, but it would be great if you find a fix for that as well if possible.

Thanks!

**Comment by devanshkansagra** [comment]
image

**Comment by devanshkansagra** [comment]
Hey maintainers, Can you review my pull request?

**Comment by Spiral-Memory** [comment]
Hey @devanshkansagra 

I watched your video, it's looking good ! However in recursive quoting, attachment is not getting displayed right ? 

I'll review the code to see if it's not breaking anything.. 

Thanks a lot for the contribution !

**Comment by devanshkansagra** [comment]
> Hey @devanshkansagra
> 
> I watched your video, it's looking good ! However in recursive quoting, attachment is not getting displayed right ?
> 
> I'll review the code to see if it's not breaking anything..
> 
> Thanks a lot for the contribution !

it is getting displayed

**Comment by Spiral-Memory** [comment]
> > Hey @devanshkansagra
> > 
> > I watched your video, it's looking good ! However in recursive quoting, attachment is not getting displayed right ?
> > 
> > I'll review the code to see if it's not breaking anything..
> > 
> > Thanks a lot for the contribution !
> 
> This issue is there with the main RC, hence it is not able to display the attachment while recursive quoting

Ohh Okay ! 

Let me see it, I'll let you know

**Comment by devanshkansagra** [comment]
Sure

File: packages/react/src/views/AttachmentHandler/Attachment.js [hunk 1]
```diff
@@ -14,6 +14,8 @@ const Attachment = ({ attachment, host, type, variantStyles = {} }) => {
         attachment={attachment}
         host={host}
         variantStyles={variantStyles}
+        authorIcon={attachment.author_icon}
```
Comment (Spiral-Memory) [inline packages/react/src/views/AttachmentHandler/Attachment.js:17]: Send only one prop, i.e., `author` or `authorInfo`, and destructure where required. If possible, try not to send extra props. If required, send `author`.

File: packages/react/src/views/AttachmentHandler/Attachment.js [hunk 2]
```diff
@@ -44,6 +51,42 @@ const Attachment = ({ attachment, host, type, variantStyles = {} }) => {
       />
     );
   }
+  if (attachment && attachment.attachments[0]?.image_url) {
```
Comment (Spiral-Memory) [inline packages/react/src/views/AttachmentHandler/Attachment.js:54]: Why introduce redundant rendering of a component? Instead, you can add the required condition where we are already using `ImageAttachment`, `AudioAttachment`, and `VideoAttachment`.

File: packages/react/src/views/AttachmentHandler/AudioAttachment.js [hunk 3]
```diff
@@ -1,18 +1,135 @@
...
+  attachment,
+  host,
+  type,
+  authorIcon,
+  authorName,
+  variantStyles,
+}) => {
+  const { RCInstance } = useContext(RCContext);
+  const { colors } = useTheme();
+  const getUserAvatarUrl = (icon) => {
```
Comment (Spiral-Memory) [inline packages/react/src/views/AttachmentHandler/AudioAttachment.js:18]: prop name such as authorUrl here will be better.

File: packages/react/src/views/AttachmentHandler/TextAttachment.js [hunk 4]
```diff
@@ -12,11 +12,6 @@ const TextAttachment = ({ attachment, type, variantStyles = {} }) => {
     return URL;
   };
 
-  let attachmentText = attachment?.text;
```
Comment (Spiral-Memory) [inline packages/react/src/views/AttachmentHandler/TextAttachment.js:15]: Why you removed this.. this is needed when you try to render a instruction message sent by RC app

File: packages/react/src/views/QuoteMessage/QuoteMessage.js [hunk 5]
```diff
@@ -10,18 +10,71 @@ import {
...
+    const URL = `${instanceHost}/avatar/${username}`;
     return URL;
   };
   const styles = useQuoteMessageStyles();
   const setQuoteMessage = useMessageStore((state) => state.setQuoteMessage);
 
   const { classNames, styleOverrides } = useComponentOverrides('QuoteMessage');
+
+  // Function to render the message content or attachment preview
+  const renderMessageContent = () => {
```
Comment (Spiral-Memory) [inline packages/react/src/views/QuoteMessage/QuoteMessage.js:28]: Let's not have a function here like this, try using JSX with required conditions

Comment (devanshkansagra) [reply packages/react/src/views/QuoteMessage/QuoteMessage.js:28]: Okay will modify this

File: packages/react/src/views/AttachmentHandler/Attachment.js (hunk 1 above)
Comment (devanshkansagra) [reply packages/react/src/views/AttachmentHandler/Attachment.js:17]: Okay will change this

File: packages/react/src/views/AttachmentHandler/Attachment.js (hunk 2 above)
Comment (devanshkansagra) [reply packages/react/src/views/AttachmentHandler/Attachment.js:54]: It was easier for me to implement like this and this is also a conditional based rendering

File: packages/react/src/views/AttachmentHandler/AudioAttachment.js (hunk 3 above)
Comment (devanshkansagra) [reply packages/react/src/views/AttachmentHandler/AudioAttachment.js:18]: Okay will change this

**Review by Spiral-Memory — APPROVED** [review]

**Comment by Spiral-Memory** [comment]
Thank you for contribution @devanshkansagra

**Pull request merged**